src/overlay/OverlayManagerImpl.cpp

```diff
@@ -1115,4 +1118,153 @@ OverlayManagerImpl::updateFloodRecord(StellarMessage const& oldMsg,
     ZoneScoped;
     mFloodGate.updateRecord(oldMsg, newMsg);
 }
+
+void
+OverlayManagerImpl::demand()
+{
+    ZoneScoped;
+    if (mShuttingDown || mWaitingToDemand)
+    {
+        return;
+    }
+    auto const& authenticatedPeers = getAuthenticatedPeers();
```

Comment:
this list needs to filter out peers that do not have pull mode enabled

Reply:
Fixed.